### ISSUE TITLE
Use webstorm Require.js Plugin?

### ISSUE THREAD

**Issue by zepumph** [phetsims/chipper#784]
I was curious what this got for us, so I decided to do a bit of investigation. The plugin is very cool in that it supports "go to declaration" via require statements. That was the main goal of this exploration. Here is what I did.

* Download "Require.js Plugin"
* Enabled it
* Go to Settings -> Other Settings -> Require.js Plugin Settings
* Give it a root and path to config, here is what I did:
  ![image](https://user-images.githubusercontent.com/6856943/62493371-65e7dd00-b77d-11e9-8d4e-d3794f28b6a3.png)

Notice that many paths now support ctrl+b ("go to declaration" command)

There are two concerns of mine:

* This doesn't work 100%. For example BRAND isn't correct, because it needs a global: `    BRAND: '../../brand/' + phet.chipper.brand + '/js',`. Also for an un explained reason `SCENERY_PHET` wouldn't work. All others did though that I saw. When one doesn't support it highlights it red, and marks the whole file with red underlining (like eslint plugin too). This is a pain for false negatives. 
* I couldn't get it to support our plugins out of the box, and I'm unsure how to add that support.
* It is annoying that the config is only good for a single sim. I could imagine that this may be a nice starting point for a new developer, because they could then get things setup just in the sim they were working on, but for me this is a deal breaker. A solution to this though would be to create an `all-config.js` based on all repos with js code in it.  This would have every path in the project in it. It could live in chipper or perennial or phet-info/ide/idea to support all paths. 

I wanted to note my investigation in this issue, and bring it up with devs to see if this could be desirable to investigate further. Would people use this in its current state? Would people use it if the above problems could be solved? Did anyone encounter any other problems with it? The last commit on the plugin was in 2016, and it only has 81 stars (https://github.com/Fedott/WebStormRequireJsPlugin). Let's discuss!

**Comment by pixelzoom** [comment]
For anyone interested, the plugin page is https://plugins.jetbrains.com/plugin/7337-require-js-plugin.

I'm not clear on how this works.  @zepumph what does 'supports "go to declaration" via require statements' mean?

**Comment by pixelzoom** [comment]
Not valuable for me. Shift-Command-N works reliably for me.

**Comment by zepumph** [comment]
I gave a demo in developer meeting today. The main issues seem to be around the plugin support. We (@samreid) talked a bit about investigate our own cross platform logic, like a grunt task that given a require url, it tells you what file and what line it is in. Then we could add that to sublime/webstorm in a thin layer to get the ide to go to that file/line.

We ended up discussing that if we are going to pour time into this problem, likely we should look into `import` statements instead of requirejs, but that is a huge investment. 

I'm probably not going to look into this plugin at all. Closing for now.
